HHS/Head-Start-TTADP
# [TTAHUB-949] Fix Goal Delete

## Description of change
- The Goal context menu to Delete or Edit a goal on an AR should now have 'Remove' NOT 'Delete'. If a goal is removed from an AR both the Goals and Objectives created should remain in the DB. But should no longer be associated to the report.
- Fixes issue where creating two goals on an AR and deleting the first. It would delete the last goal and leave the page in a broken state.

## How to test

- Create a goal with both new and existing Goals. Removing these Goals from the report and saving should only remove the association. The Goals and Objectives tables should still have both Goals and Objectives.
- Create multiple goals on the AR and delete the first one. Only the first goal should be deleted and you should be able to delete other goals after.

All other Goal functionality on an AR should work as it did before.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-949


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
